http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/GkeAppConfig.scala

```diff
@@ -0,0 +1,48 @@
+package org.broadinstitute.dsde.workbench.leonardo.config
+
+import org.broadinstitute.dsde.workbench.google2.KubernetesSerializableName.ServiceAccountName
+import org.broadinstitute.dsde.workbench.leonardo.{Chart, KubernetesService, ServiceConfig, ServiceId}
+import org.broadinstitute.dsp.{ChartName, ChartVersion}
+
+sealed trait GkeAppConfig {
+  def chartName: ChartName
+  def chartVersion: ChartVersion
+  def releaseNameSuffix: String
+  def namespaceNameSuffix: String
+
+  def chart: Chart = Chart(chartName, chartVersion)
+  def kubernetesServices: List[KubernetesService]
+}
+
+final case class GalaxyAppConfig(releaseNameSuffix: String,
+                                 chartName: ChartName,
+                                 chartVersion: ChartVersion,
+                                 namespaceNameSuffix: String,
+                                 services: List[ServiceConfig],
+                                 serviceAccount: ServiceAccountName,
+                                 uninstallKeepHistory: Boolean,
+                                 postgresPassword: String,
+                                 orchUrl: String,
+                                 drsUrl: String)
+    extends GkeAppConfig {
+  override lazy val kubernetesServices: List[KubernetesService] = services.map(s => KubernetesService(ServiceId(-1), s))
```

Comment:
I guess the simple answer is, brings something that was happening somewhere else into this class.

Syntax wise, it's a lazy val (so gets evaluated the first time it's asked for, then remembered), which takes the services section from config and turns each of them into a KubernetesService instance, with an ID of `-1`. 

In terms of why, I'd defer to an IA member, but I guess implicitly it assumes that Galaxy and Cromwell configs only have a single service (at least in the leo config), otherwise the IDs would clash. And later on, this is used to calculate the proxy URLs

Reply:
The ID of -1 was particularly confusing to me-- I'd love to hear more about that from IA folks.

Reply:
yea a bit confusing, but this is just a helper function for persisting `KubernetesService` to DB...`id` is an auto-incrementing key, and the way slick works is you just stub out the id when saving it to DB